enhancements/testdata-separation.md

```diff
@@ -0,0 +1,109 @@
+---
+title: testdata-separation
+authors:
+  - "@theishshah"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-07-12
+last-updated: 2021-07-20
+status: TBD
+---
+
+# testdata-separation
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The current bundle includes both an operator as well as its testdata. This results in inflated binaries and scenarios where users may want to release an operator bundle sans testdata but are unable to do so. The solution to this issue is to provide users the ability to use and generate a slim bundle which is decoupled from the testing necessities. Due to the nature of the testdata, it needs the rest of the bundle so it has limited value alone.
+
+## Motivation
+
+The motivation for this EP is to provide users with the option to deliver a slim bundle without testdata.
+* Creates a smaller size bundle to ship
+* Prevents including testing resources in shipped bundle
+  * Stop potential elevated RBAC's in the test bundle from being included in production environments
+
+### Goals
+
+The goal is to create a solution which allows developers to continue to access and utilize the testdata while still having the option to ship an operator without said testdata in the bundle.
+
+### Non-Goals
+
+The goal of this EP is not to change the contents of the bundle nor the testdata, but just to reorganize them, either with multiple bundles or by allowing the tools to modify a bundle to achieve the desired state.
+
+## Proposal
+
+Please note that this proposed solution was elevated from the Alternatives selection after discussions, and the previous implementation has been moved down.
+
+Hard Split
+* Split existing bundle in a testdata only bundle paired with a operator essentials only bundle
+  * Allows the bundle to save significant space/reduce its size
+  * Allows operator developers to securely ship with including potentially sensitive testdata
```

Comment:
Will scorecard permit multiple testdata bundles to be referenced from an operator-only bundle? Consider the case where the operator authors decide some of their tests are sensitive, but other tests (e.g. the SDK provided tests) are not.

If multiple are allowed, can the main bundle include a scorecard configuration (like they do now) AND reference another bundle? 

If multiple are not allowed, what will we tell operator authors who want to separate tests out like this?

Reply:
Another question. If the operator bundle is in a public registry repo and the testdata bundle is in a private registry repo, will callers of `operator-sdk scorecard` be expected to configure the credentials for the private testadata bundle repo out-of-band?

Should public operator bundles even reference private testdata bundles? If that is an intended use case, will SDK handle unauthorized errors gracefully? We can't assume that callers of `scorecard` will know in advance a) whether the main bundle includes its test config and b) if not, whether the referenced test config image is publicly available (or at least available to anyone who can pull the bundle itself).

Reply:
@joelanford that is an EXCELLENT question because that's definitely one we have with `run bundle`.